src/utils.js

```diff
@@ -131,9 +131,9 @@ export const randomId      = ()                   => Math.random().toString(36).
 /**
  * TODO: use generators instead? (ie. "yield")
  */
-export const iMax = (iVal, max = -Infinity) => (iVal > max ? iVal : max); 
-export const iMin = (iVal, min = +Infinity) => (iVal < min ? iVal : min);
-export const iAvg = (iVal, avg = 0, idx = 1) => (iVal + avg * (idx - 1)) / idx;
+export const iMax = (iVal, max = -Infinity) =>  (((typeof (iVal) === 'number') && (iVal > max)) ? iVal : max);
+export const iMin = (iVal, min = +Infinity) =>  (((typeof (iVal) === 'number') && (iVal < min)) ? iVal : min);
+export const iAvg = (iVal, avg = 0, idx = 1) => (((typeof (iVal) === 'number') && idx > 1) ? ((iVal + avg * (idx - 1)) / idx) : ((iVal && idx < 2) ? iVal : avg));
```

Comment:
I hope you will agree with me that in the long run it is almost impossible to remember the reason behind all these checks...

Reply:
The test `typeof (iVal) === 'number'` is needed because this is not always the case. I remember that this occurs when both speed and pace are processed. I have to work more with comments. You are right as always. (I am not a programmer).

`iAvg`  can I change to
```
export const iAvg = (iVal, avg = 0, idx = 1) => (typeof (iVal) === 'number') ? (iVal + avg * (idx - 1)) / idx : undefined;
```
But it needs some testing.

Reply:
I can't. Then `avgspeed` and `avgpace` are 0.